docs/rpyc.rst

```diff
@@ -1,87 +1,114 @@
-   like
+   .. code-block:: py
 
-   .. code-block:: none
+       import rpyc
 
-      INFO:SLAVE/18812:server started on [0.0.0.0]:18812
+       # Create a RPyC connection to the remote ev3dev device.
+       # Use the hostname or IP address of the ev3dev device.
+       # If this fails, verify your IP connectivty via ``ping X.X.X.X``
+       conn = rpyc.classic.connect('X.X.X.X')
 
-   and keep running.
+       # import ev3dev2 on the remote ev3dev device
+       ev3dev2_motor = conn.modules['ev3dev2.motor']
+       ev3dev2_sensor = conn.modules['ev3dev2.sensor']
+       ev3dev2_sensor_lego = conn.modules['ev3dev2.sensor.lego']
 
-3. Now you are ready to connect to the RPyC server from your desktop PC. The
-   following python script should make a large motor connected to output port
-   ``A`` spin for a second.
+       # Use the LargeMotor and TouchSensor on the remote ev3dev device
+       motor = ev3dev2_motor.LargeMotor(ev3dev2_motor.OUTPUT_A)
+       ts = ev3dev2_sensor_lego.TouchSensor(ev3dev2_sensor.INPUT_1)
 
-   .. code-block:: py
+       # If the TouchSensor is pressed, run the motor
+       while True:
+           ts.wait_for_pressed()
+           motor.run_forever(speed_sp=200)
 
-       import rpyc
-       conn = rpyc.classic.connect('ev3dev') # host name or IP address of the EV3
-       ev3 = conn.modules['ev3dev2.ev3']      # import ev3dev2.ev3 remotely
-       m = ev3.LargeMotor('outA')
-       m.run_timed(time_sp=1000, speed_sp=600)
-
-You can run scripts like this from any interactive python environment, like
-ipython shell/notebook, spyder, pycharm, etc.
-
-Some *advantages* of using RPyC with ev3dev are:
-
-* It uses much less resources than running ipython notebook on EV3; RPyC server
-  is lightweight, and only requires an IP connection to the EV3 once set up (no
-  ssh required).
-* The scripts you are working with are actually stored and edited on your
-  desktop PC, with your favorite editor/IDE.
-* Some robots may need much more computational power than what EV3 can give
-  you. A notable example is the Rubics cube solver: there is an algorithm that
-  provides almost optimal solution (in terms of number of cube rotations), but
-  it takes more RAM than is available on EV3. With RPYC, you could run the
+           ts.wait_for_released()
+           motor.stop()
+
+
+Pros
+====
+* The RPyC server is lightweight and only requires an IP connection (no ssh required).
+* Some robots may need much more computational power than an EV3 can give
+  you. A notable example is the Rubik's cube solver. There is an algorithm that
+  provides an almost optimal solution (in terms of number of cube rotations), but
+  it takes more RAM than is available on EV3. With RPyC, you could run the
   heavy-duty computations on your desktop.
 
+Cons
+====
 The most obvious *disadvantage* is latency introduced by network connection.
 This may be a show stopper for robots where reaction speed is essential.
```

Comment:
Also requires full python3 rather than micropython

Reply:
You know there is part of me that thinks about everything we support via micropython now and how much faster it starts...should we just drop the RPyC angle and push MQTT as “the” daisy chain solution? It is would be less confusing in that there would only be one choice AND it would work for both python3 and micropython.

Reply:
I wouldn't be opposed to that. I do like the MQTT approach if we figure out a good way to package it up/describe it. I started replying to the other thread but got distracted and will finish.

Reply:
yeah lets see what shakes out of this mqtt/daisy-chain brainstorming.